client/metrics.go

```diff
@@ -17,111 +17,180 @@ package pd
-	cmdFailedDurationGetRegionByID            = cmdFailedDuration.WithLabelValues("get_region_byid")
-	cmdFailedDurationScanRegions              = cmdFailedDuration.WithLabelValues("scan_regions")
-	cmdFailedDurationGetStore                 = cmdFailedDuration.WithLabelValues("get_store")
-	cmdFailedDurationGetAllStores             = cmdFailedDuration.WithLabelValues("get_all_stores")
-	cmdFailedDurationUpdateGCSafePoint        = cmdFailedDuration.WithLabelValues("update_gc_safe_point")
+	cmdDurationScatterRegion = cmdDuration.WithLabelValues("scatter_region")
+	cmdDurationScatterRegions = cmdDuration.WithLabelValues("scatter_regions")
+	cmdDurationGetOperator = cmdDuration.WithLabelValues("get_operator")
+	cmdDurationSplitRegions = cmdDuration.WithLabelValues("split_regions")
+	cmdDurationSplitAndScatterRegions = cmdDuration.WithLabelValues("split_and_scatter_regions")
+	cmdDurationLoadKeyspace = cmdDuration.WithLabelValues("load_keyspace")
+	cmdDurationUpdateKeyspaceState = cmdDuration.WithLabelValues("update_keyspace_state")
+	cmdDurationGet = cmdDuration.WithLabelValues("get")
+	cmdDurationPut = cmdDuration.WithLabelValues("put")
+
+	cmdFailDurationGetRegion = cmdFailedDuration.WithLabelValues("get_region")
+	cmdFailDurationTSO = cmdFailedDuration.WithLabelValues("tso")
+	cmdFailDurationGetAllMembers = cmdFailedDuration.WithLabelValues("get_member_info")
+	cmdFailDurationGetPrevRegion = cmdFailedDuration.WithLabelValues("get_prev_region")
+	cmdFailedDurationGetRegionByID = cmdFailedDuration.WithLabelValues("get_region_byid")
+	cmdFailedDurationScanRegions = cmdFailedDuration.WithLabelValues("scan_regions")
+	cmdFailedDurationGetStore = cmdFailedDuration.WithLabelValues("get_store")
+	cmdFailedDurationGetAllStores = cmdFailedDuration.WithLabelValues("get_all_stores")
+	cmdFailedDurationUpdateGCSafePoint = cmdFailedDuration.WithLabelValues("update_gc_safe_point")
 	cmdFailedDurationUpdateServiceGCSafePoint = cmdFailedDuration.WithLabelValues("update_service_gc_safe_point")
-	cmdFailedDurationLoadKeyspace             = cmdDuration.WithLabelValues("load_keyspace")
-	cmdFailedDurationUpdateKeyspaceState      = cmdDuration.WithLabelValues("update_keyspace_state")
-	requestDurationTSO                        = requestDuration.WithLabelValues("tso")
-	cmdFailedDurationGet                      = cmdFailedDuration.WithLabelValues("get")
-	cmdFailedDurationPut                      = cmdFailedDuration.WithLabelValues("put")
-)
+	cmdFailedDurationLoadKeyspace = cmdDuration.WithLabelValues("load_keyspace")
+	cmdFailedDurationUpdateKeyspaceState = cmdDuration.WithLabelValues("update_keyspace_state")
+	requestDurationTSO = requestDuration.WithLabelValues("tso")
+	cmdFailedDurationGet = cmdFailedDuration.WithLabelValues("get")
+	cmdFailedDurationPut = cmdFailedDuration.WithLabelValues("put")
+}
 
-func init() {
+// RegisterMetrics registers all metrics for PD client.
+func RegisterMetrics() {
```

Comment:
Where do we call this function?

Reply:
from tidb, I guess

Reply:
yesh, the specific location should be [here](https://github.com/tidbcloud/tidb-cse/blob/keyspace-on-release-6.6/metrics/metrics.go#L294-L296) like client-go

Reply:
yesh, the specific location should be [here](https://github.com/tidbcloud/tidb-cse/blob/keyspace-on-release-6.6/metrics/metrics.go#L294-L296) like client-go

Reply:
The behavior seems to have changed since you modified the `init()` function. Will this cause any issues for those who do not wish to register explicitly?

Reply:
No, if users want to register `pd_client` metrics, these should call `Register` explicitly.

Reply:
My point is that you have changed the default behavior of initializing metrics. For some libraries that are already using it, upgrading may cause some problems because they may not be aware of this change.